### PR TITLE
move to go modules

### PR DESCRIPTION
**Pull request by 06kellyjac**
# Changes

Moved to go modules rather than dep. I guess could use git submodules instead :shrug:

Also bumped the dependency. The contents in the vendor dir doesn't match the commit in https://github.com/tektoncd/catalog/blob/master/Gopkg.lock#L10 (`51c53a31e7791871544c7a6526b6e0ad6ce25d65`) of https://github.com/tektoncd/plumbing/blob/51c53a31e7791871544c7a6526b6e0ad6ce25d65/scripts/library.sh#L328-L336
So I thought it best to just fully update it.
Also the latest version of plumbing scripts doesnt use dep either

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by chmouel** [comment]
sounds good thnaks

/lgtm

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module github.com/tektoncd/catalog
+
+go 1.16
```
Comment (06kellyjac) [inline go.mod:3]: @chmouel
This looks like what's causing the failure

Shall I move this down to `go 1.13` like plumbing or shall I use `go 1.15` as that seems to be what's available:

```
>> go version
go version go1.15.2 linux/amd64
```

https://storage.googleapis.com/tekton-prow/pr-logs/pull/tektoncd_catalog/663/pull-tekton-catalog-build-tests/1371435036477231105/build-log.txt

Or should I look at bumping the go version used to `go 1.16`?

Comment (chmouel) [reply go.mod:3]: let's keep the same as what we have in plumbing?

Comment (06kellyjac) [reply go.mod:3]: sure

Comment (06kellyjac) [reply go.mod]: moved to `1.13`

**Comment by vdemeester** [comment]
Huh interesting, I don't know why there is the `dep` files in there even. I think the `plumbing` part was added/update kind-of manually 😅

**Review by vdemeester — APPROVED** [review]

**Comment by 06kellyjac** [comment]
I can swap it to a script or a git submodule, or just document it as a thing to be updated manually if youd prefer

---

If its fine as is its probably worth checking the version change fixed the test failure

**Comment by 06kellyjac** [comment]
Ok that's looking better although it's logging out `./deps.go:2:8: imported and not used: "github.com/tektoncd/plumbing/scripts"` because obviously it doesnt use anything.
If that log is just a warning rather than a breaking error I guess we can just ignore it.

File: deps.go [hunk 2]
```diff
@@ -0,0 +1,2 @@
+package deps
+import "github.com/tektoncd/plumbing/scripts"
```
Comment (vdemeester) [inline deps.go:2]: `import _ "github.com/tektoncd/plumbing/scripts"` to remove the compilation error.

See https://github.com/tektoncd/pipeline/blob/main/hack/tools.go for examples of "this" in pipeline.

**Comment by vdemeester** [comment]
@06kellyjac well `gofmt` ain't happy 😓 🤦🏼 
```
-------------------------------------------
---- Checking go code style with gofmt ----
-------------------------------------------
diff -u deps.go.orig deps.go
--- deps.go.orig	2021-03-18 14:12:49.801019333 +0000
+++ deps.go	2021-03-18 14:12:49.801019333 +0000
@@ -1,2 +1,3 @@
 package deps
+
 import _ "github.com/tektoncd/plumbing/scripts"
```

**Comment by 06kellyjac** [comment]
pushed the fmt fix :slightly_smiling_face:

**Pull request merged**